internal/translayer/deployment/deployment.go

```diff
@@ -0,0 +1,46 @@
+package deployment
+
+import (
+	"context"
+	"errors"
+
+	"go.mongodb.org/atlas/mongodbatlas"
+	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/translayer"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/controller/atlas"
+)
+
+type Service struct {
```

Comment:
what is a `Service`?

Reply:
We could revisit naming, but a Service is a particular interface of a feature, you have a service to handle users, another for deployments, etc. The client is a separate entity that allows you to access and user services.

Reply:
Let's make this crystal clear in the godoc then as the term `Service` has a much broader meaning than `Client`.

Reply:
While I agree this might need a rename, lets consider that "Service" here is always qualified by the package name so in reality is `deployment.Service` here or `dbuser.Service` on the other layer.

Maybe I could rename as `Deployments` and `Users` but keep in mind it will still be a bit of stuttering as `deployment.Deployments` and `dbuser.Users`.

Reply:
coming in next update: I opted for `Atlas{resource plural}Service` for the interface and `ProductionAtlas{resource plural}` for the concrete type.